File: Intents/StartActivity/src/main/java/co/edu/udea/compumovil/intents/MainActivity.java
package co.edu.udea.compumovil.intents;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private EditText editText;
    public static final String EXTRA_MESSAGE = "co.edu.udea.compumovil.intents.MESSAGE";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editText = (EditText) findViewById(R.id.edit_message);
    }

    /**
     * Called when the user clicks the Send button
     */
    public void sendMessage(View view) {
        String msg = editText.getText().toString();
        editText.setText("");
        Intent intent = new Intent(this, DisplayMessageActivity.class);
        intent.putExtra(EXTRA_MESSAGE, msg);
        startActivity(intent);
    }
}

File: Activities/StartActivity/src/main/java/co/edu/udea/compumovil/startactivity/MainActivity.kt
package co.edu.udea.compumovil.startactivity

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.content.Intent
import android.util.Log


class MainActivity : AppCompatActivity() {

    val EXTRA_MESSAGE = "co.edu.udea.compumovil.startactivity.MESSAGE"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    /** Called when the user taps the Send button */
    fun sendMessage(view: View) {
        val editText = findViewById<EditText>(R.id.editText)
        val message = editText.text.toString()
        Log.d("TAG", message)
        val intent = Intent(this, DisplayMessageActivity::class.java).apply {
            putExtra(EXTRA_MESSAGE, message)
        }
        startActivity(intent)
    }


}
